app/jobs/calendar_importer/calendar_importer.rb

```diff
@@ -16,19 +16,34 @@ class InaccessibleFeed < StandardError; end
     CalendarImporter::Parsers::Ticketsolve
   ].freeze
 
+  MANUAL_PARSERS = (DETECTABLE_PARSERS.dup.concat [
+    CalendarImporter::Parsers::LdJson
+    # CalendarImporter::Parsers::RssFeed
+  ]).freeze
+
   def initialize(calendar)
     @calendar = calendar
 
     validate_feed!
   end
 
   def parser
-    @parser ||=
-      if @calendar.importer_mode == 'auto'
-        PARSERS.find { |parser| parser.handles_url?(@calendar.source) }
-      else
-        PARSERS.find { |parser| parser::KEY == @calendar.importer_mode }
+    return @parser if @parser
+
+    if @calendar.importer_mode == 'auto'
+      @parser = DETECTABLE_PARSERS.find { |parser| parser.handles_url?(@calendar.source) }
+
+      if @parser.blank?
+        # TODO: now try to detect ld+json
```

Comment:
I think this `TODO` is done by the implementation. By attempting to use it we validate if it's valid.